### PR TITLE
Docs/doc 95/doc 112/technical term store

### PR DESCRIPTION
**Pull request by Rachel-Reverie**
Please annotate your PR title to describe what the PR does, then give a brief bulleted description of your PR below. PR titles should begin with [BUGFIX], [FEATURE],  [DOCS], or [MAINTENANCE]. If a new feature introduces breaking changes for the Great Expectations API or configuration files, please also add [BREAKING]. You can read about the tags in our [contributor checklist](https://docs.greatexpectations.io/en/latest/contributing/contribution_checklist.html).

Changes proposed in this pull request:
- Adds technical term page for Store
- Adds technical term page for Checkpoint Store
- Adds technical term page for Evaluation Parameter Store
- Adds technical term page for Metric Store
- Adds technical term page for Data Docs Store
- Adds technical term page for Validation Results Store
- Adds technical term page for Expectation Store

After submitting your PR, CI checks will run and @ge-cla-bot will check for your CLA signature.

For a PR with nontrivial changes, we review with both design-centric and code-centric lenses.

In a design review, we aim to ensure that the PR is consistent with our relationship to the open source community, with our software architecture and abstractions, and with our users' needs and expectations. That review often starts well before a PR, for example in github issues or slack, so please link to relevant conversations in notes below to help reviewers understand and approve your PR more quickly (e.g. `closes #123`).

### Definition of Done
Please delete options that are not relevant.

- [X] I have made corresponding changes to the documentation

Thank you for submitting!

### CODE REVIEW

**Comment by AFineDayFor** [comment]
For the included content, lgtm.
We can also use a query_store.
for example [1](https://greatexpectationstalk.slack.com/messages/CUTCNHN82/p1641313262016500) and [2](https://greatexpectationstalk.slack.com/messages/CUTCNHN82/p1637254778456300?thread_ts=1637251464.452700&cid=CUTCNHN82); I wouldn't say it's blocking us/this being added; so much as it's an area we can add at some point.

Thoughts?

**Review by AFineDayFor — APPROVED** [review]
So long as you're okay with my thought on focusing on query_stores at some point in the future good to go; if you'd like to add it before-hand I'll be around.

File: docs/terms/store.md
Comment (joshua-stauffer) [inline docs/terms/store.md]: could these be links?

File: docs/terms/store.md
Comment (joshua-stauffer) [inline docs/terms/store.md]: either remove this section or rename the header 😄

File: docs/terms/data_docs_store.md
Comment (joshua-stauffer) [inline docs/terms/data_docs_store.md]: should 'Human' be capitalized here?

File: docs/terms/data_docs_store.md
Comment (joshua-stauffer) [inline docs/terms/data_docs_store.md]: This last sentence slowed me down a bit -- i wonder if something like the following may be easier to parse: 

"This update will only include the Validation Results from the Checkpoint itself and will not re-render all of your existing Data Docs. If a Checkpoint does not include an `UpdateDataDocsAction`, the Validation Results will not be rendered into Data Docs."

File: docs/terms/data_docs_store.md
Comment (joshua-stauffer) [inline docs/terms/data_docs_store.md]: this note is not rendering correctly -- i'm at a loss for why, but maybe try adding an extra \n?

File: docs/terms/store.md
Comment (Rachel-Reverie) [reply docs/terms/store.md]: Yes.  In fact, they are supposed to be.  I must've missed going back to them to provide the links after I added those pages.

File: docs/terms/data_docs_store.md
Comment (joshua-stauffer) [inline docs/terms/data_docs_store.md:54]: Is `data_context.build_data_docs()` worth mentioning as part of the API?

File: docs/terms/data_docs_store.md
Comment (Rachel-Reverie) [reply docs/terms/data_docs_store.md]: It's case sensitive.  I should've used a lowercase n.

File: docs/terms/metric_store.md
Comment (joshua-stauffer) [inline docs/terms/metric_store.md:34]: It might be useful to either include an example config or link to another page that has more details on setting up a metric store

File: docs/terms/metric_store.md
Comment (joshua-stauffer) [inline docs/terms/metric_store.md:45]: `StoreMetricsAction` Action -- seems redundant, could we drop the second Action?

Comment (joshua-stauffer) [reply docs/terms/metric_store.md:45]: ..but if that's the pattern we're using elsewhere, better to be consistent 😄

**Review by joshua-stauffer — APPROVED** [review]
looks great -- i caught a couple of typos and left some suggestions, but once you take a look at them feel free to merge!

File: docs/terms/store.md
Comment (talagluck) [inline docs/terms/store.md]: Could you explain "Evaluation Parameter Stores (or Metric Stores)" in a bit more detail?  It looks like the page on Metric Stores distinguishes the two, so it seems contradictory to equate them here, and differentiate them there.

File: docs/terms/store.md
Comment (Rachel-Reverie) [reply docs/terms/store.md]: Haha.  Thanks for the catch.  Clearly I meant to go back to this page after creating the other ones and then... forgot. >.>

File: docs/terms/data_docs_store.md
Comment (Rachel-Reverie) [reply docs/terms/data_docs_store.md]: It should not.  Fixed.

File: docs/terms/data_docs_store.md
Comment (Rachel-Reverie) [reply docs/terms/data_docs_store.md]: I like that better as well.  Changed the content to match your suggestion.

File: docs/terms/data_docs_store.md
Comment (Rachel-Reverie) [reply docs/terms/data_docs_store.md:54]: I added it in.

File: docs/terms/metric_store.md
Comment (Rachel-Reverie) [reply docs/terms/metric_store.md:34]: Link added.

File: docs/terms/metric_store.md
Comment (Rachel-Reverie) [reply docs/terms/metric_store.md:45]: I agree that it looks odd, but technically `StoreMetricsAction` is the name of an Action, so in this case Action is the subject noun for the sentence.

File: docs/terms/store.md
Comment (Rachel-Reverie) [reply docs/terms/store.md]: That is in fact a typo.  I had received some documentation when I was consolidating sources that implied the two were functionally the same.  Later I received some clarification that they are not due to how each stores the information they contain; I updated the doc but must have overlooked this spot that had them conflated still.  I've split them apart now.

**Pull request merged**
